### DISCUSSION TITLE
No reasonable sim with ngspice

### DISCUSSION THREAD

**Discussion by hauserulrich** [ra3xdh/qucs_s#317] [Q&A] [answered]
Hello,

new here, so don't be too harsh :-).
Migrated from Qucs 0.0.19 (Win & Linux) to Qucs-s 2.0.0. Both systems do not giv me reasonable sim results in time or frequency domain. A simple RL with battery yields "Note: No compatibility mode selected!", no further error messages, but the results are rubbish on Windows and on Linux. The result looks like my starting condition (0A at the L) are not recognised by the simulator.
I couldn't find much of sense about the above note. It might be the source of my trouble but I don't know how to cure it. Qucs with the same schematic yielded good results. Creating the schematic again in Qucs-s didn't help either.
Hints would be much appreciated
Ulrich

**Comment by ra3xdh** [comment]
Please submit a schematic. I didn't observe the issues with relays before. The relay has SPICE model.

**Comment by ra3xdh** [comment, marked as answer]
Also be sure to set `InitialDC=no` in TRAN simulation properties to apply initial conditions.

**Comment by hauserulrich** [comment]
Thanks, it was the InitialDC. But that definitely changed from Qucs. But it then is misleading to be able to set the initial current at the Inductance itself as this conflicts with the initial DC check. Then, InitialDC shoulf be no as a default, isn't it?

**Comment by ra3xdh** [comment]
`InitailDC=no` should not be set as default. Default behavior is to derive the initial conditions from OP simulation. The `InitialDC=no` applies the `UIC` option for Ngspice that tells simulator use the IC defined by user. Probably QUcsator can inspect the netlist and apply the option automatically if the IC is specified somewhere.

**Comment by hauserulrich** [comment]
It seems to be an advantage sometimes, sometimes not. I agree there might be more cases were initial DC makes sense. There is no point arguing that. In the netlist, I0 is given. I do not know ngspice, but the netlist shows IC=0 to indicate that the inductance has no current at t=0. So it could be used by ngspice, but obviously it isn't.
Baseline: I know how to deal with it. Thank you for the help. It always good to learn.
